app/assets/javascripts/controllers/FileCtrl.coffee

```diff
@@ -43,37 +44,71 @@
     return
 
   ###*
-   * save file as bruse_file and append bruse_file to the file
-   * @param {obj} file  the file
+   * save file to temp array
+   * @param {obj} file  the file to add
   ###
   $scope.add = (file) ->
+    if file.is_dir and !file.contents
```

Comment:
Not really, it checks if the file is **a** folder/directory and then makes sure that the folder's contents has been loaded.